dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitorInternalMethodsTest.kt

```diff
@@ -0,0 +1,1401 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.monitor
+
+import android.util.Log
+import com.datadog.android.Datadog
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.log.internal.user.UserInfo
+import com.datadog.android.log.internal.user.UserInfoProvider
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumResourceKind
+import com.datadog.android.rum.assertj.RumEventAssert.Companion.assertThat
+import com.datadog.android.rum.internal.domain.RumEvent
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.android.utils.forge.exhaustiveAttributes
+import com.datadog.tools.unit.extensions.ApiLevelExtension
+import com.datadog.tools.unit.forge.aThrowable
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyZeroInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.LongForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.util.UUID
+import kotlin.system.measureNanoTime
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(ApiLevelExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DatadogRumMonitorInternalMethodsTest {
+
+    lateinit var testedMonitor: DatadogRumMonitor
+
+    @Mock
+    lateinit var mockWriter: Writer<RumEvent>
+
+    @Mock
+    lateinit var mockTimeProvider: TimeProvider
+
+    @Mock
+    lateinit var mockUserInfoProvider: UserInfoProvider
+
+    lateinit var mockedUserInfo: UserInfo
+
+    @Forgery
+    lateinit var fakeApplicationId: UUID
+
+    @LongForgery(min = 0L)
+    var fakeTimestamp: Long = 0L
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        Datadog.setVerbosity(Log.VERBOSE)
+        GlobalRum.updateApplicationId(fakeApplicationId)
+        mockedUserInfo = UserInfo(
```

Comment:
There's already a `UserInfoForgeryFactory` so you can create a property like this : 

```
    @Forgery
    lateinit var fakeUserInfo : UserInfo
```